Enigma crashes when importing a server jar and exporting the source code

Enigma, when used with any of the jars produced by yarn itself, runs fine, but when importing a server and using the same mappings, it crashes with the following error message:
`java.lang.Error: java.util.EmptyStackException
	at cuchaz.enigma.gui.dialog.ProgressDialog.lambda$runInThread$0(ProgressDialog.java:66)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.util.EmptyStackException
	at java.util.Stack.peek(Stack.java:102)
	at java.util.Stack.pop(Stack.java:84)
	at cuchaz.enigma.bytecode.translators.TranslationSignatureVisitor.visitInnerClassType(TranslationSignatureVisitor.java:30)
	at org.objectweb.asm.signature.SignatureReader.parseType(SignatureReader.java:198)
	at org.objectweb.asm.signature.SignatureReader.accept(SignatureReader.java:122)
	at cuchaz.enigma.translation.representation.Signature.remap(Signature.java:63)
	at cuchaz.enigma.translation.representation.Signature.translate(Signature.java:96)
	at cuchaz.enigma.translation.MappingTranslator.translate(MappingTranslator.java:22)
	at cuchaz.enigma.translation.representation.entry.ClassDefEntry.translate(ClassDefEntry.java:80)
	at cuchaz.enigma.translation.representation.entry.ClassDefEntry.translate(ClassDefEntry.java:24)
	at cuchaz.enigma.translation.representation.entry.ParentedEntry.translate(ParentedEntry.java:62)
	at cuchaz.enigma.translation.MappingTranslator.translate(MappingTranslator.java:22)
	at cuchaz.enigma.Deobfuscator.lambda$deobfuscateClasses$3(Deobfuscator.java:213)
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
	at java.util.HashMap$KeySpliterator.forEachRemaining(HashMap.java:1553)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
	at java.util.stream.ReduceOps$ReduceTask.doLeaf(ReduceOps.java:747)
	at java.util.stream.ReduceOps$ReduceTask.doLeaf(ReduceOps.java:721)
	at java.util.stream.AbstractTask.compute(AbstractTask.java:316)
	at java.util.concurrent.CountedCompleter.exec(CountedCompleter.java:731)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinTask.doInvoke(ForkJoinTask.java:401)
	at java.util.concurrent.ForkJoinTask.invoke(ForkJoinTask.java:734)
	at java.util.stream.ReduceOps$ReduceOp.evaluateParallel(ReduceOps.java:714)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:233)
	at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:499)
	at cuchaz.enigma.Deobfuscator.deobfuscateClasses(Deobfuscator.java:228)
	at cuchaz.enigma.Deobfuscator.writeSources(Deobfuscator.java:195)
	at cuchaz.enigma.gui.GuiController.lambda$exportSource$2(GuiController.java:132)
	at cuchaz.enigma.gui.dialog.ProgressDialog.lambda$runInThread$0(ProgressDialog.java:64)
	... 1 more`
I was using the same version, snapshot 19w13a, for both mappings and the server jar. Here is the server jar: https://launcher.mojang.com/v1/objects/c63080e9349640fda5820bbe48cacc623c99c496/server.jar

I'm having this bug as well. Here is the recent stacktrace, with build 163:
```
java.lang.Error: java.util.EmptyStackException
	at cuchaz.enigma.gui.dialog.ProgressDialog.lambda$0(ProgressDialog.java:71)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.util.EmptyStackException
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at java.util.concurrent.ForkJoinTask.getThrowableException(ForkJoinTask.java:598)
	at java.util.concurrent.ForkJoinTask.reportException(ForkJoinTask.java:677)
	at java.util.concurrent.ForkJoinTask.invoke(ForkJoinTask.java:735)
	at java.util.stream.ReduceOps$ReduceOp.evaluateParallel(ReduceOps.java:714)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:233)
	at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:499)
	at cuchaz.enigma.EnigmaProject.exportRemappedJar(EnigmaProject.java:163)
	at cuchaz.enigma.gui.GuiController.lambda$5(GuiController.java:191)
	at cuchaz.enigma.gui.dialog.ProgressDialog.lambda$0(ProgressDialog.java:67)
	... 1 more
Caused by: java.util.EmptyStackException
	at java.util.Stack.peek(Stack.java:102)
	at java.util.Stack.pop(Stack.java:84)
	at cuchaz.enigma.bytecode.translators.TranslationSignatureVisitor.visitInnerClassType(TranslationSignatureVisitor.java:30)
	at org.objectweb.asm.signature.SignatureReader.parseType(SignatureReader.java:198)
	at org.objectweb.asm.signature.SignatureReader.accept(SignatureReader.java:122)
	at cuchaz.enigma.translation.representation.Signature.remap(Signature.java:63)
	at cuchaz.enigma.translation.representation.Signature.translate(Signature.java:96)
	at cuchaz.enigma.translation.MappingTranslator.translate(MappingTranslator.java:22)
	at cuchaz.enigma.translation.representation.entry.ClassDefEntry.translate(ClassDefEntry.java:75)
	at cuchaz.enigma.translation.representation.entry.ClassDefEntry.translate(ClassDefEntry.java:1)
	at cuchaz.enigma.translation.representation.entry.ParentedEntry.translate(ParentedEntry.java:70)
	at cuchaz.enigma.translation.representation.entry.ParentedEntry.translate(ParentedEntry.java:1)
	at cuchaz.enigma.translation.MappingTranslator.translate(MappingTranslator.java:22)
	at cuchaz.enigma.translation.mapping.EntryRemapper.deobfuscate(EntryRemapper.java:75)
	at cuchaz.enigma.ProposingTranslator.translate(ProposingTranslator.java:29)
	at cuchaz.enigma.EnigmaProject.lambda$0(EnigmaProject.java:150)
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
	at java.util.HashMap$KeySpliterator.forEachRemaining(HashMap.java:1553)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
	at java.util.stream.ReduceOps$ReduceTask.doLeaf(ReduceOps.java:747)
	at java.util.stream.ReduceOps$ReduceTask.doLeaf(ReduceOps.java:721)
	at java.util.stream.AbstractTask.compute(AbstractTask.java:316)
	at java.util.concurrent.CountedCompleter.exec(CountedCompleter.java:731)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
```

same problem in 0.16.0 build.171 running under Java 8
when exporting source for the server of 1.14.4
`java.lang.Error: java.util.EmptyStackException
	at cuchaz.enigma.gui.dialog.ProgressDialog.lambda$runOffThread$0(ProgressDialog.java:72)
	at java.base/java.lang.Thread.run(Thread.java:832)
Caused by: java.util.EmptyStackException
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:500)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:481)
	at java.base/java.util.concurrent.ForkJoinTask.getThrowableException(ForkJoinTask.java:603)
	at java.base/java.util.concurrent.ForkJoinTask.reportException(ForkJoinTask.java:678)
	at java.base/java.util.concurrent.ForkJoinTask.invoke(ForkJoinTask.java:737)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateParallel(ReduceOps.java:919)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:233)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
	at cuchaz.enigma.EnigmaProject.exportRemappedJar(EnigmaProject.java:172)
	at cuchaz.enigma.gui.GuiController.lambda$exportSource$5(GuiController.java:211)
	at cuchaz.enigma.gui.dialog.ProgressDialog.lambda$runOffThread$0(ProgressDialog.java:68)
	... 1 more
Caused by: java.util.EmptyStackException
	at java.base/java.util.Stack.peek(Stack.java:101)
	at java.base/java.util.Stack.pop(Stack.java:83)
	at cuchaz.enigma.bytecode.translators.TranslationSignatureVisitor.visitInnerClassType(TranslationSignatureVisitor.java:30)
	at org.objectweb.asm.signature.SignatureReader.parseType(SignatureReader.java:198)
	at org.objectweb.asm.signature.SignatureReader.accept(SignatureReader.java:122)
	at cuchaz.enigma.translation.representation.Signature.remap(Signature.java:63)
	at cuchaz.enigma.translation.representation.Signature.translate(Signature.java:96)
	at cuchaz.enigma.translation.MappingTranslator.translate(MappingTranslator.java:22)
	at cuchaz.enigma.translation.representation.entry.ClassDefEntry.translate(ClassDefEntry.java:75)
	at cuchaz.enigma.translation.representation.entry.ClassDefEntry.translate(ClassDefEntry.java:23)
	at cuchaz.enigma.translation.representation.entry.ParentedEntry.translate(ParentedEntry.java:70)
	at cuchaz.enigma.translation.representation.entry.ParentedEntry.translate(ParentedEntry.java:24)
	at cuchaz.enigma.translation.MappingTranslator.translate(MappingTranslator.java:22)
	at cuchaz.enigma.translation.mapping.EntryRemapper.deobfuscate(EntryRemapper.java:75)
	at cuchaz.enigma.ProposingTranslator.translate(ProposingTranslator.java:29)
	at cuchaz.enigma.EnigmaProject.lambda$exportRemappedJar$0(EnigmaProject.java:159)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.HashMap$KeySpliterator.forEachRemaining(HashMap.java:1694)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ReduceOps$ReduceTask.doLeaf(ReduceOps.java:952)
	at java.base/java.util.stream.ReduceOps$ReduceTask.doLeaf(ReduceOps.java:926)
	at java.base/java.util.stream.AbstractTask.compute(AbstractTask.java:327)
	at java.base/java.util.concurrent.CountedCompleter.exec(CountedCompleter.java:746)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1016)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1665)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1598)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:177)
`

Still reproducable, 0.18 build 186
AdoptOpenJDK 8.0.252.09-hotspot

```java.lang.Error: java.util.EmptyStackException
        at cuchaz.enigma.gui.dialog.ProgressDialog.lambda$runOffThread$0(ProgressDialog.java:72)
        at java.lang.Thread.run(Thread.java:748)
Caused by: java.util.EmptyStackException
        at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
        at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
        at java.util.concurrent.ForkJoinTask.getThrowableException(ForkJoinTask.java:598)
        at java.util.concurrent.ForkJoinTask.reportException(ForkJoinTask.java:677)
        at java.util.concurrent.ForkJoinTask.invoke(ForkJoinTask.java:735)
        at java.util.stream.ReduceOps$ReduceOp.evaluateParallel(ReduceOps.java:714)
        at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:233)
        at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:566)
        at cuchaz.enigma.EnigmaProject.exportRemappedJar(EnigmaProject.java:184)
        at cuchaz.enigma.gui.GuiController.lambda$exportSource$4(GuiController.java:189)
        at cuchaz.enigma.gui.dialog.ProgressDialog.lambda$runOffThread$0(ProgressDialog.java:68)
        ... 1 more
Caused by: java.util.EmptyStackException
        at java.util.Stack.peek(Stack.java:102)
        at java.util.Stack.pop(Stack.java:84)
        at cuchaz.enigma.bytecode.translators.TranslationSignatureVisitor.visitInnerClassType(TranslationSignatureVisitor.java:30)
        at org.objectweb.asm.signature.SignatureReader.parseType(SignatureReader.java:198)
        at org.objectweb.asm.signature.SignatureReader.accept(SignatureReader.java:122)
        at cuchaz.enigma.translation.representation.Signature.remap(Signature.java:63)
        at cuchaz.enigma.translation.representation.Signature.translate(Signature.java:96)
        at cuchaz.enigma.translation.MappingTranslator.translate(MappingTranslator.java:22)
        at cuchaz.enigma.translation.representation.entry.ClassDefEntry.translate(ClassDefEntry.java:75)
        at cuchaz.enigma.translation.representation.entry.ClassDefEntry.translate(ClassDefEntry.java:23)
        at cuchaz.enigma.translation.representation.entry.ParentedEntry.translate(ParentedEntry.java:69)
        at cuchaz.enigma.translation.representation.entry.ParentedEntry.translate(ParentedEntry.java:23)
        at cuchaz.enigma.translation.MappingTranslator.translate(MappingTranslator.java:22)
        at cuchaz.enigma.translation.mapping.EntryRemapper.deobfuscate(EntryRemapper.java:82)
        at cuchaz.enigma.translation.ProposingTranslator.translate(ProposingTranslator.java:27)
        at cuchaz.enigma.EnigmaProject.lambda$exportRemappedJar$1(EnigmaProject.java:171)
        at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
        at java.util.HashMap$KeySpliterator.forEachRemaining(HashMap.java:1556)
        at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:482)
        at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:472)
        at java.util.stream.ReduceOps$ReduceTask.doLeaf(ReduceOps.java:747)
        at java.util.stream.ReduceOps$ReduceTask.doLeaf(ReduceOps.java:721)
        at java.util.stream.AbstractTask.compute(AbstractTask.java:327)
        at java.util.concurrent.CountedCompleter.exec(CountedCompleter.java:731)
        at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
        at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
        at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
        at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)```

--- a/enigma/src/main/java/cuchaz/enigma/bytecode/translators/TranslationSignatureVisitor.java
+++ b/enigma/src/main/java/cuchaz/enigma/bytecode/translators/TranslationSignatureVisitor.java
@@ -1,10 +1,11 @@
 package cuchaz.enigma.bytecode.translators;
 
-import cuchaz.enigma.Enigma;
-import org.objectweb.asm.signature.SignatureVisitor;
-
 import java.util.Stack;
 import java.util.function.Function;
+
+import org.objectweb.asm.signature.SignatureVisitor;
+
+import cuchaz.enigma.Enigma;
 
 public class TranslationSignatureVisitor extends SignatureVisitor {
 	private final Function<String, String> remapper;
@@ -31,6 +32,7 @@ public class TranslationSignatureVisitor extends SignatureVisitor {
 		if (!name.startsWith(lastClass+"$")){//todo see if there's a way to base this on whether there were type params or not
 			name = lastClass+"$"+name;
 		}
+		classStack.push(name);
 		String translatedEntry = this.remapper.apply(name);
 		if (translatedEntry.contains("/")){
 			translatedEntry = translatedEntry.substring(translatedEntry.lastIndexOf("/")+1);

